vispy/scene/widgets/grid.py

```diff
@@ -31,6 +31,21 @@ def __init__(self, spacing=6, **kwargs):
         self.spacing = spacing
         self._n_added = 0
         self._default_class = ViewBox  # what to add when __getitem__ is used
+        self._solver = None
+        self._need_solver_recreate = True
+
+        # width and height of the Rect used to place child widgets
+        self.var_w = Variable("w_rect")
+        self.var_h = Variable("h_rect")
+        self.var_x = Variable("x_rect")
+        self.var_y = Variable("x_rect")
```

Comment:
Yep, I know :) It simplified life to implement it as such, so I was doing it that way. Since the x and y positions were purely equations without multiple constraints, it's as easy as taking the x and y constraint equations and just using them as code